Angular/Angular guidelines and best practices/File and module organization and naming.md

```diff
@@ -15,53 +15,113 @@ Try to keep the code grouped by domain. For example, if you have a `Post` class,
 
 ```
 src/app/
-├── components/
-│   └── post/
-└── pages/
-    ├── users-container/
-    └── posts-container/
-```
-
-- `PostsCount` component is used only inside the `users-container` top-level route, but it is used in both `user-details-container` and `users-list-container` sub-routes:
-
-```
-src/app/
-└── pages/
-    └── users-container/
-        ├── components/
-        │   └── posts-count/
-        ├── user-details-container/
-        └── users-list-container/
+├── components/ # these components are used in both posts and users feature modules
+│   ├── post-overview/ # renders basic info about a post (without full content)
+│   └── user-avatar/ # renders username and profile image
+├── services/ # these services are used in both posts and users feature modules
+│   ├── auth/ # session management, not important for this example
+│   ├── user/
+│   │   ├── user.service.ts # has methods for data fetching
+│   │   ├── user.service.spec.ts
+│   │   ├── user.testing.service.ts
+│   │   ├── user.interface.ts
+│   │   └── user.model.ts
+│   └── post/ # basically the same as user.service
+├── pages/
+│   ├── users/
+│   │   ├── pages/
+│   │   │   ├── users-index/
+│   │   │   │   ├── components/
+│   │   │   │   │   └── users-list/ # renders a list of user-avatar components
+│   │   │   │   ├── users-index.component.ts # renders some header/footer and users-list component
+│   │   │   │   ├── users-index-routing.module.ts # eagerly-loads users-index.component on /
+│   │   │   │   └── users-index.module.ts # imports users-index-routing.module
+│   │   │   └── user-index/
+│   │   │       ├── components/
+│   │   │       │   └── user-details/ # renders user-avatar component, additional user info and a list of user's posts (post-overview)
+│   │   │       ├── user-index.component.ts # renders some header/footer and users-details component
+│   │   │       ├── user-index-routing.module.ts # eagerly-loads user-index.component on /
+│   │   │       └── user-index.module.ts # imports user-index-routing.module
+│   │   ├── users-routing.module.ts # lazy-loads users-list.module on /, user-details.module on /:id/view
+│   │   └── users.module.ts # imports users-routing.module
+│   └── posts/
+│       ├── components/ # these components are used only in posts feature/domain module
+│       │   ├── post-form/ # form used for both editing and creation
+│       │   └── post-content/ # renders post content
+│       ├── services/
+│       │   └── social-sharing/ # service that is used only inside posts module
+│       ├── pages/
+│       │   ├── posts-index/
+│       │   │   ├── components/
+│       │   │   │   └── posts-list/ # renders post-overview and user-avatar component for each post
+│       │   │   ├── posts-index.component.ts # renders some header/footer and posts-list component
+│       │   │   ├── posts-index-routing.module.ts # eagerly-loads posts-index.component on /
+│       │   │   └── posts-index.module.ts # imports posts-index-routing.module
```

Comment:
Should `posts-intex.testing.module.ts` and `posts-index.mock.component.ts` also be here?

Reply:
No, because these components are loaded via routing (not via template of some other component) so you do not need to import mock/testing versions of those components in tests for some other componen.